recipes/libpq/all/conanfile.py

```diff
@@ -93,9 +94,15 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def generate(self):
+        self._patch_sources()
```

Comment:
patch in build() please

Reply:
The migration guide says that access to dependencies should be done in the generate() function
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html
> It is also very important to know that every access to the information from the dependencies must be done in the generate() method using the [self.dependencies access](https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface).

Reply:
it also works in build(), and conan-center recommends to patch in build()